symphony-ws/src/main/java/se/havochvatten/symphony/web/ReportREST.java

```diff
@@ -158,6 +181,38 @@ public Response getComparisonReport(@Context HttpServletRequest req,
         // Image is available at /calculation/comparison/{a}/{b}. (computed on the fly, and cached client-side)
     }
 
+    @GET
+    @Path("/comparison/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RolesAllowed("GRP_SYMPHONY")
+    @ApiOperation(value = "Return data for differential scenario report using an implicit baseline calculation",
+            response = ComparisonReportResponseDto.class)
+    public Response getComparisonReport(@Context HttpServletRequest req,
+                                        @PathParam("id") int scenarioId,
+                                        @DefaultValue("") @QueryParam("lang") String preferredLanguage,
+                                        @DefaultValue("false") @QueryParam("reverse") boolean reverse,
+                                        @Context UriInfo uriInfo) {
+        logger.info("Comparing report "+scenarioId+" with baseline");
+        try {
+            var scenarioRes =
+                CalcUtil.getCalculationResultFromSessionOrDb(scenarioId, req.getSession(), calcService)
+                    .orElseThrow(javax.ws.rs.BadRequestException::new);
+
+            var implicitResult = calcService.getImplicitBaselineCalculation(scenarioRes);
+
+            if (hasAccess(scenarioRes, req.getUserPrincipal()))
```

Comment:
Move above row 201 to return as early as possible?

Reply:
Sure, that's better 👍